src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs

```diff
@@ -387,163 +413,194 @@ ELSE NULL
+
+                        table.Columns.Add(column);
                     }
                 }
             }
         }
 
-        void GetIndexes(
-            NpgsqlConnection connection,
-            IReadOnlyList<DatabaseTable> tables,
-            string tableFilter,
-            List<uint> constraintIndexes)
+        /// <summary>
+        /// Queries the database for defined indexes and registers them with the model.
+        /// </summary>
+        /// <param name="connection">The database connection.</param>
+        /// <param name="tables">The database tables.</param>
+        /// <param name="tableFilter">The table filter fragment.</param>
+        /// <param name="constraintIndexes">The constraint indexes.</param>
+        /// <param name="logger">The diagnostic logger.</param>
+        static void GetIndexes(
+            [NotNull] NpgsqlConnection connection,
+            [NotNull] IReadOnlyList<DatabaseTable> tables,
+            [CanBeNull] string tableFilter,
+            [NotNull] List<uint> constraintIndexes,
+            [NotNull] IDiagnosticsLogger<DbLoggerCategory.Scaffolding> logger)
         {
-            using (var command = connection.CreateCommand())
-            {
-                var getColumnsQuery = @"
+            var commandText = $@"
 SELECT
-    idxcls.oid AS idx_oid, nspname, cls.relname AS cls_relname, idxcls.relname AS idx_relname, indisunique, indkey, amname,
-    CASE WHEN indexprs IS NULL THEN NULL ELSE pg_get_expr(indexprs, cls.oid) END AS exprs,
-    CASE WHEN indpred IS NULL THEN NULL ELSE pg_get_expr(indpred, cls.oid) END AS pred
+  idxcls.oid AS idx_oid,
+  nspname,
+  cls.relname AS cls_relname,
+  idxcls.relname AS idx_relname,
+  indisunique,
+  indkey,
+  amname,
+  CASE
+    WHEN indexprs IS NULL THEN NULL
+    ELSE pg_get_expr(indexprs, cls.oid)
+  END AS exprs,
+  CASE
+    WHEN indpred IS NULL THEN NULL
+    ELSE pg_get_expr(indpred, cls.oid)
+  END AS pred
 FROM pg_class AS cls
 JOIN pg_namespace AS ns ON ns.oid = cls.relnamespace
 JOIN pg_index AS idx ON indrelid = cls.oid
 JOIN pg_class AS idxcls ON idxcls.oid = indexrelid
 JOIN pg_am AS am ON am.oid = idxcls.relam
 WHERE
-    cls.relkind = 'r' AND
-    nspname NOT IN ('pg_catalog', 'information_schema')
-AND
-    NOT indisprimary
-" + tableFilter;
+  cls.relkind = 'r' AND
+  nspname NOT IN ('pg_catalog', 'information_schema') AND
+  NOT indisprimary AND
+  cls.relname <> '{HistoryRepository.DefaultTableName}'
+  {tableFilter}";
 
-                command.CommandText = getColumnsQuery;
+            using (var command = new NpgsqlCommand(commandText, connection))
+            using (var reader = command.ExecuteReader())
+            {
+                var tableGroups = reader.Cast<DbDataRecord>().GroupBy(ddr => (
+                    tableSchema: ddr.GetValueOrDefault<string>("nspname"),
+                    tableName: ddr.GetValueOrDefault<string>("cls_relname")
+                ));
```

Comment:
Or rewrite the previously commented `GroupBy` like here (:

Reply:
Of the three options, I think this is a nice balance of compactness and readability. I'm going to go with this, but bring those trailing parentheses up to the end of the previous line.